hasher-matcher-actioner/hmalib/common/timebucketizer.py

```diff
@@ -0,0 +1,161 @@
+        self.storage_path = storage_path
+        self.id = id
+        self.type = type
+
+        self.buffer_size = buffer_size
+        self.buffer: t.List[T] = []
+
+
+    def _calculate_bucket_endpoints(self, time: datetime.datetime, bucket_width: datetime.timedelta):
+        now = time
+        rounded = now - (now - datetime.datetime.min) % bucket_width
+        return (rounded, rounded + self.bucket_width)
+
+    def add_record(self, record: T) -> None:
+        """
+        Adds the record to the current bucket.
+        """
+        if len(self.buffer) >= MAX_BUFFER_SIZE or self.end <= datetime.datetime.now():
+            self._flush()
+        self.buffer.append(record)
+
+    def destroy_remaining(self) -> None:
+        """
+        Used to trigger a force flush of remaining data stored inside buffer
+        """
+        if not len(self.buffer):
+            return
+        self._flush()
+
+    # Change back to t.List[T] after other method is implemented
+    def get_records(self) -> t.List[str]:
+        """
+        Used for testing. Returns a list of all filepathways found starting from the storage_path
+        """
+        directory_path = os.path.join(self.storage_path, self.type, "")
+
+        list_of_pathways = [
+            # Leaving this here for now: Hashrecord.from_csv(val)
+            val
+            for sublist in [
+                [os.path.join(i[0], j) for j in i[2]] for i in os.walk(directory_path)
+            ]
+            for val in sublist
+        ]
+
+        return list_of_pathways
+
+    def _flush(self) -> None:
+        """
+        Flushes the current data onto storage source
+        """
+        accurate_date = os.path.join(
+            str(self.start.year),
+            str(self.start.month),
+            str(self.start.day),
+            str(self.start.hour),
+            str(self.start.minute),
+        )
+
+        file_name = str(self.id) + ".csv"
+        directory_path = os.path.join(self.storage_path, self.type, accurate_date)
+        file_path = os.path.join(directory_path, file_name)
+
+        if not os.path.isdir(directory_path):
+            os.makedirs(directory_path)
+
+        # Opening in append mode because it's possible we have to write to the same file multiple times with preexisting data
+        # newline is overriden to prevent spaces in the csv file between sections
+        with open(file_path, "a+", newline="") as outfile:
+            writer = csv.writer(outfile)
+            writer.writerows(map(lambda x: x.to_csv(), self.buffer))
+
+        self.start, self.end = self._calculate_bucket_endpoints(datetime.datetime.now(), self.bucket_width)
+        self.buffer = []
+
+    def get_file_contents(self, since: datetime.datetime, until: datetime.datetime, type: str, storage_path: str, bucket_width: datetime.timedelta, type_class: t.Type[CSViable]):
+        """
+        Returns all the data stored inside csv file given the file_path
```

Comment:
Docstring is wrong. No file_path is given. It does not return "all" data. Fix this.